### ISSUE TITLE
Downloader not working with Flutter 3.7 background isolates

### ISSUE THREAD

**Issue by kuhnroyal** [fluttercommunity/flutter_downloader#809]
**Describe the bug**

`FlutterDownloader.initialize` calls `WidgetsFlutterBinding.ensureInitialized()` which fails in the new background isolates with `UI actions are only available on root isolate.`.

See https://medium.com/flutter/introducing-background-isolate-channels-7a299609cad8

**To Reproduce**

```dart
import ‘package:flutter/services.dart’;
import ‘flutter_downloader/flutter_downloader.dart’;

void main() {
 // Identify the root isolate to pass to the background isolate.
 // (API introduced in Flutter 3.7)
 RootIsolateToken rootIsolateToken = RootIsolateToken.instance!;
 Isolate.spawn(_isolateMain, rootIsolateToken);
}

void _isolateMain(RootIsolateToken rootIsolateToken) async {
 // Register the background isolate with the root isolate.
 BackgroundIsolateBinaryMessenger
   .ensureInitialized(rootIsolateToken);
 // You can now use the plugin
  await FlutterDownloader.initialize(); ///
}
```

**Expected behavior**

On Flutter 3.7 `initialize` should not call `WidgetsFlutterBinding.ensureInitialized()` if it is called in a background isolate.

**Device information:**

- plugin version: v1.10.1+2

**Comment by markst** [comment]
is this the same as #740 or #736 ?

**Comment by kuhnroyal** [comment]
> is this the same as #740 or #736 ?

Not this is specifically related to the newly introduced Flutter 3.7 feature.

**Comment by nashihu** [comment]
i'm using Flutter 3.7.2 and follow [this comment](https://github.com/fluttercommunity/flutter_downloader/issues/807#issuecomment-1415989716) after upgrading from 3.3.10 , it works for me so far

**Comment by kuhnroyal** [comment]
> i'm using Flutter 3.7.2 and follow [this comment](https://github.com/fluttercommunity/flutter_downloader/issues/807#issuecomment-1415989716) after upgrading from 3.3.10 , it works for me so far

Not related

**Comment by bartekpacia** [comment]
This was fixed in #816 and released in [v1.10.2](https://pub.dev/packages/flutter_downloader/changelog#1102).
